pkg/config/remote/service/tracer_cache.go

```diff
@@ -0,0 +1,114 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```

Comment:
There's already a few caching libraries imported in the build of the agent. As an example https://github.com/patrickmn/go-cache, which seem to support expirations. Given that nothing here is going to end up in a hot path `interface{}` are not an issue either.
WDYT ?

Reply:
Mhmm, this particular one doesn't have the behavior I think we need. For once, I didn't find a way to specify a max size of the cache and that feels dangerous. Also the API is not the same as what we have now. There is explicit replacements instead of automatic replacement. I can rename the struct because cache might sound confusing. Essentially this struct is the `TracerTracker`. Skimming the code I didn't really find other caches that fit the bill (saw `groupcache` in `go.mod` and some util that uses the cache you recommended).

Reply:
👍🏻